### ISSUE TITLE
Disable filteredRelationCount feature in prisma > 4.16.0

### ISSUE THREAD

**Issue by erikmellum** [prisma/prisma#20383]
### Bug description

Our team upgraded from 4.14.1 to 5.0.0 and ran into bugs due to our usage of typegraphql-prisma and the new default enabled `filteredRelationCount` feature released to GA in 4.16.0.

This feature results in autogenerated types containing many new required `getters` that we do not need. Is there a way to disable this?

See the related issue we opened on typegraphql-prisma: https://github.com/MichalLytek/typegraphql-prisma/issues/408

### How to reproduce

Upgrade to 4.16.0 or beyond

### Expected behavior

I want to be able to disable filteredRelationCount

### Prisma information

Prisma 5.0.0

### Environment & setup

- OS: macOS
- Database: MySQL
- Node.js version: `18.12.1`

### Prisma Version

```
5.0.0
```

**Comment by janpio** [comment]
Generally, we have no plans to disable any functionality we added, unless it is incorrect or broken. But if I understand correctly, this is doing exactly what we want it to - via `typegraphql-prisma` this just leads to more generated code than before? Isn't this more a configuration for `typegraphql-prisma` to skip generating these getters that you do not need?

**Comment by erikmellum** [comment]
@janpio I would agree that the feature is working correctly. I can't quite tell at this point if:

1. we need to modify `typegraphql-prisma` to accommodate for the new code
2. maintain our types manually without using `typegraphql-prisma`
3. flag to disable this feature in `prisma`
4. ensure that the autogenerated getters using the new `filteredRelationCount` feature are marked as optional in the autogenerated type (I anticipate this would be a change in `typegraphql-prisma`)

For our team, having typescript types autogenerated using `typegraphql-prisma` for each of the models in the `schema.prisma` is a big value add. I'll be happy to provide our solution when we get there, but until then, we are blocked on moving from `4.14.1` to prisma >= `4.16.0`.

I appreciate the response.

**Comment by janpio** [comment]
We talked about this internally, and don't really see a viable path to implement this in Prisma itself. 

The challenge here is that all these features are interconnected. If you disable one, even if you do it really cleanly, in theory all other features can break as well. Disable another one, same situation again. So to confidently ship something like this we would need to test Prisma Client with each feature disabled individually, and then also with all combinations 😨 We already run into a similar problem with our preview feature flags that _add_ functionality, and had to do quite a dance to cover that.

So unfortunately for now our suggestion is that this is implemented on a deeper level instead by Prisma itself.

We'll keep this issue open though (with a slightly renamed title) to collect further feedback and especially use cases. Maybe this is actually a super common problem, and we should _find_ a way to make this possible if there are enough people asking and making a good case for it.

**Comment by erikmellum** [comment]
@janpio thanks for discussing with the team. It seems reasonable to just leave this open for feedback. Hopefully it is not a common issue. Solving it in `typegraphql-prisma` or in peoples own codebase is reasonable. I did successfully test two possible solutions for this. These are relevant for others who may come across this 

1. Run a post:generate script to clean up the types that typegraphql-generated to no longer include the getters. I did a quick and dirty version that fixed the issues that we ran into. We did not opt to use this approach since it felt brittle, but it was certainly the smallest amount of effort.

```javascript
const path = './node_modules/@generated/type-graphql/resolvers/outputs'
console.log('Processing all generated *Count files to remove getters.')
try {
  ;(async () => {
    // eslint-disable-next-line @typescript-eslint/no-var-requires
    const fs = require('fs')
    const files = await fs.readdirSync(path)
    files.forEach(async (file) => {
      if (file.includes('Count')) {
        const filePath = `${path}/${file}`
        const fileContent = await fs.readFileSync(filePath, { encoding: 'utf8' })
        const newContent = fileContent.replace(/get(.+?);/g, '')
        fs.writeFileSync(filePath, newContent)
      }
    })
    console.log('Successfully processed all generated *Count files to remove getters.')
  })()
} catch (err) {
  console.error('Error processing generated *Count files to remove errors', err)
}

```
2. Replace `include: { _count: true }` with `_count: { select: xyzRelation: true }` in addition to type coercion after the prisma operation. ex:

```
 return (await context.prisma.xyzModel.findMany({
        include: {
          _count: {
            select: {
              xyzRelation: true,
            },
          },
        },
      })) as xyzModel[]
```

For this solution we had to refactor anywhere we were using _count: true in a prisma operation. We also had to do type coercion (which feels a little bit wrong) in a number of places. 

Probably the best solution is the one we did not pursue yet.

3. Patch typegraphql-prisma to make these `get*` functions optional in the type. We may investigate this option further if we see runtime issues due to our type coercion solution.
